wordpress-mobile/WordPress-MediaPicker-Android
# [Compile Warnings As Errors] Resolve Warnings & Enable All Warnings as Errors

This PR resolves/suppresses all warnings for all modules and then enables all warnings as errors as the default for this project (a3e3911eca425bda0919535cc035689cf150ca28).

PS: This is done just to restrict further Kotlin related warnings to creep into this repo and with it to also enable future Kotlin updates, for example see below warning that would have made the Kotlin `1.7` update a bit more in the future:

`Non exhaustive 'when' statements on enum will be prohibited in 1.7, add 'XYZ' branches or 'else' branch instead`

-----

Warnings Suppression List:

1. [Suppress all deprecated warnings for mediapicker](https://github.com/wordpress-mobile/WordPress-MediaPicker-Android/commit/e530a9726db491b2723c7ac24a7df03b1278c597)

Test Restructure List:

1. [Remove test check from buildkite](https://github.com/wordpress-mobile/WordPress-MediaPicker-Android/commit/54e697120b508aa5afb34a7382066aef7c30d523)

Dependency Versions Refactor List:

1. [Extract all sdk version to a separate ext block](https://github.com/wordpress-mobile/WordPress-MediaPicker-Android/commit/baa9b6979b12b6052bf6adabb184cd5a51e6c584)
2. [Remove unnecessary build tools version configuration](https://github.com/wordpress-mobile/WordPress-MediaPicker-Android/commit/166de801a2c39a208358295fa465a1c9c8f26281)
3. [Remove unnecessary extra dagger version configuration](https://github.com/wordpress-mobile/WordPress-MediaPicker-Android/commit/a572bab5b197de7008dd2bca616c35d886bc6d07)
4. [Remove unused hilt version](https://github.com/wordpress-mobile/WordPress-MediaPicker-Android/commit/447ba495bc7cd97bf32b83cbe731284f9c34fd61)
5. [Rename lifecycle to androidx lifecycle version](https://github.com/wordpress-mobile/WordPress-MediaPicker-Android/commit/c7e0ba26da6f31fd2ea86f3ff6460b1708e0d0d1)
6. [Rename datastore to androidx datastore version](https://github.com/wordpress-mobile/WordPress-MediaPicker-Android/commit/c15473cfb5f72ac08d70ee9cb50d198a66a09b08)
7. [Rename swipe refresh layout to androidx swipe to refresh version](https://github.com/wordpress-mobile/WordPress-MediaPicker-Android/commit/2bfadeb9b7b0fb75f057cd249c9580cc6825abad)
8. [Rename constraint layout to androidx constraint layout version](https://github.com/wordpress-mobile/WordPress-MediaPicker-Android/commit/df7e1cfcba66ceed5b65bd2b41ed958715da7a61)
9. [Rename appcompat to androidx appcompat version](https://github.com/wordpress-mobile/WordPress-MediaPicker-Android/commit/dd4e18b24b871cc4debe5ce14d8cf9a94da854ec)
10. [Rename core to androidx core version](https://github.com/wordpress-mobile/WordPress-MediaPicker-Android/commit/348603954e852b20c1eb50f261f35838806bb553)
11. [Rename coroutines to kotlinx coroutines version](https://github.com/wordpress-mobile/WordPress-MediaPicker-Android/commit/4123f858850210fe0fe5683619b5856b412f44e9)
12. [Rename material to google material version](https://github.com/wordpress-mobile/WordPress-MediaPicker-Android/commit/99cf6603105e1d5e3066403da596487a1e99e0ce)
13. [Rename photoview to chrisbanes photoview version](https://github.com/wordpress-mobile/WordPress-MediaPicker-Android/commit/49f965184052cfa6ce72acb50364516dbdae0797)
14. [Rename navigation fragment to androidx navigation version](https://github.com/wordpress-mobile/WordPress-MediaPicker-Android/commit/86610a73eaea41752e1a9181c85f7b35561db5ff)
15. [Rename retrofit to squareup retrofit version](https://github.com/wordpress-mobile/WordPress-MediaPicker-Android/commit/bacd3e2fa5ca841c00a229a127bc4bc905b27495)
16. [Move wordpress utils version to different version group for libs](https://github.com/wordpress-mobile/WordPress-MediaPicker-Android/commit/90785d3cb686fdd9335bdd87dc763c478d16ccd9)
17. [Rename fluxc to wordPress fluxc version](https://github.com/wordpress-mobile/WordPress-MediaPicker-Android/commit/64bcf4ad2a2b3e6dcc2b8a48939029a7ef0fc5d6)
18. [Add versions groups to root build gradle](https://github.com/wordpress-mobile/WordPress-MediaPicker-Android/commit/179b6fce9bb19aeceeacfb489dea30a0fa4a2e28)

-----

## To test

1. There is nothing much to test here (if not all, most of the deprecated warnings were suppressed).
2. Verifying that all the CI checks are successful should be enough.
3. However, if you want to be thorough about reviewing this change, you could quickly smoke test the `sampleapp` app and see if it is working as expected.
